### PR TITLE
fix: #1988 tightened router based searchQuery state update block

### PR DESCRIPTION
**Pull request by farhankk360**
### Summary
Salam Alaikoum,

This PR is an attempt to fix https://github.com/quran/quran.com-frontend-next/issues/1988

### Test Plan
1. Search for any word eg.. `ta` on home page search bar
2. Click on the first search result option
3. On the search page you should be able to change the search input, remove, backspace, etc..

### Screenshots

Before:
https://github.com/quran/quran.com-frontend-next/assets/26346408/1ba04708-a0c3-4bf4-9fe4-c4bcd01dc61a

After:
https://github.com/quran/quran.com-frontend-next/assets/26346408/60c0d752-0c1d-40a2-91d1-b0b51f101e6a

### CODE REVIEW

**Review by ahmedriad1 — APPROVED** [review]
Salam alykum @farhankk360,

JZK for this amazing PR! Everything looks good masha'Allah.

**Comment by ahmedriad1** [comment]
@farhankk360 Just found a bug btw, try going to https://quran-com-git-fork-farhankk360-fix-1988search-p-1246cf-qurancom.vercel.app/search?page=1&q=test&translations=131 and navigate to page 2. The results change but the paginator's state is still on page 1.

**Comment by farhankk360** [comment]
@ahmedriad1 While fixing the paginator state issue, I came across the following issues:
- filter modal input not accepting more than 1 or 2 characters
- leading comma is added to translations query string `translations=,207`

image

Also did some minor console error cleanup.
Screenshot 2023-06-19 at 12 42 33 AM

Before:
https://github.com/quran/quran.com-frontend-next/assets/26346408/ca8dab27-831a-4845-85f2-f19a52b0a6dd

After:
https://github.com/quran/quran.com-frontend-next/assets/26346408/fa0bf1ce-1a30-4a99-95ca-90916c50fb9a

File: src/components/Search/SearchResults/SearchResultItem.tsx [hunk 1]
```diff
@@ -45,15 +45,14 @@ const SearchResultItem: React.FC<Props> = ({ result, source }) => {
     <div className={styles.container}>
       <div className={styles.itemContainer}>
         <Link
+          className={styles.verseKey}
           href={getChapterWithStartingVerseUrl(result.verseKey)}
           shouldPassHref
```
Comment (ahmedriad1) [inline src/components/Search/SearchResults/SearchResultItem.tsx:50]: I think we can remove `shouldPassHref` here.

File: src/pages/search.tsx [hunk 2]
```diff
@@ -84,26 +84,36 @@ const Search: NextPage<SearchProps> = ({ translations }): JSX.Element => {
...
-      if (router.query.page) {
-        setCurrentPage(Number(router.query.page));
-      }
-      if (router.query.languages) {
-        setSelectedLanguages(router.query.languages as string);
-      }
-      if (router.query.translations) {
-        setSelectedTranslations(router.query.translations as string);
-      }
+    if (router.isReady && !isContentModalOpen) {
```
Comment (ahmedriad1) [inline src/pages/search.tsx:87]: Can we put a comment here clarifying that we don't want to focus the search input when the translation modal is open?

File: src/pages/search.tsx [hunk 3]
```diff
@@ -84,26 +84,36 @@ const Search: NextPage<SearchProps> = ({ translations }): JSX.Element => {
...
+    }
+    if (router.query.translations) {
+      setSelectedTranslations(router.query.translations as string);
+    }
+  }, [
+    router?.query?.q,
+    router?.query?.query,
+    router?.query?.page,
+    router?.query?.languages,
+    router?.query?.translations,
```
Comment (ahmedriad1) [inline src/pages/search.tsx:115]: Why do we need `?.` after `router`?

**Review by ahmedriad1 — CHANGES_REQUESTED** [review]
Awesome! JAK for the changes! It looks awesome masha'Allah, just left some small comments. I think when you resolve them we'll be able to merge insha'Allah.

**Comment by ahmedriad1** [comment]
> Let me know if you want me to add the above issues to #1988 ticket

Yes of course, feel free to update the issue you opened to include the other bug as well.

File: src/pages/search.tsx (hunk 2 above)
Comment (farhankk360) [reply src/pages/search.tsx:87]: Gotcha

File: src/pages/search.tsx (hunk 3 above)
Comment (farhankk360) [reply src/pages/search.tsx:115]: Maybe I am overprotective here. 
I will make the cleanup change.

**Review by ahmedriad1 — APPROVED** [review]
Looks good to me! JAK!

**Pull request merged**